Add elision token filter docs #7981 (#8026)
opensearch-project/documentation-website: _analyzers/token-filters/elision.md

---
layout: default
title: Elision
parent: Token filters
nav_order: 130
---

> # Elision token filter

Reviewer: ```suggestion 

# Elision token filter

```

---
layout: default
title: Elision
parent: Token filters
nav_order: 130
---

# Elision token filter

> The `elision` token filter is used to remove elided characters from words in certain languages. Elision typically occurs in languages such as French, where words are often contracted and combined with the following word, typically by omitting a vowel and replacing it with an apostrophe. 

Reviewer 2: ```suggestion
The `elision` token filter is used to remove elided characters from words in certain languages. Elision typically occurs in languages such as French, in which words are often contracted and combined with the following word, typically by omitting a vowel and replacing it with an apostrophe. 
```

> The `elision` token filter already comes preconfigured in the following [language analyzers]({{site.url}}{{site.baseurl}}/analyzers/language-analyzers/): `catalan`, `french`, `irish`, and `italian`.

Reviewer 2: ```suggestion
The `elision` token filter is already preconfigured in the following [language analyzers]({{site.url}}{{site.baseurl}}/analyzers/language-analyzers/): `catalan`, `french`, `irish`, and `italian`.
```

> The `elision` token filter already comes preconfigured in the following [language analyzers]({{site.url}}{{site.baseurl}}/analyzers/language-analyzers/): **Catalan**, **French**, **Irish** and **Italian**.

Reviewer: ```suggestion
The `elision` token filter already comes preconfigured in the following [language analyzers]({{site.url}}{{site.baseurl}}/analyzers/language-analyzers/): `catalan`, `french`, `irish`, and `italian`.
```

Reviewer: Could we link to the language analyzers (once they are merged)?

{: .note}

## Parameters

The custom `elision` token filter can be configured with the following parameters.

Parameter | Required/Optional | Data type | Description
:--- | :--- | :--- | :--- 

> `articles` | Required if `articles_path` is not configured | Array of strings Defines which articles or short words should be removed when they appear as part of an elision.

Reviewer: ```suggestion
`articles` | Required if `articles_path` is not configured | Array of strings | Defines which articles or short words should be removed when they appear as part of an elision.
```

> `articles_path` | Required if `articles` is not configured | String | Specifies path to custom list of articles that should be removed during the analysis process. 

Reviewer: ```suggestion
`articles_path` | Required if `articles` is not configured | String | Specifies the path to a custom list of articles that should be removed during the analysis process. 
```

> `articles_case` | Optional | Boolean | Specifies whether the filter is case sensitive when matching elisions. Default is `false`.

Reviewer 2: Can the filter itself be case sensitive? Or do we mean something like "Specifies whether the filter considers/applies case sensitivity when matching elisions"?

Reviewer: Technically,  yes, the filter applies case sensitivity, but I think it's understood as is 😄

> `articles_case` | Optional | Boolean | Filter is case-sensitive when it removes articles from text during analysis. Default is `false`.

Reviewer: ```suggestion
`articles_case` | Optional | Boolean | Specifies whether the filter is case sensitive when matching elisions. Default is `false`.
```

## Example

> The following example request creates a new index named `french_texts` and configures an analyzer with `french_elision` filter:

Reviewer: Should we mention the default set of French elisions (l', m', t', qu', n', s', j', d', c', jusqu', quoiqu', lorsqu', puisqu')?

> The following example request creates a new index named `french_texts` and configures an analyzer with `french_elision` filter:

Reviewer: ```suggestion
The default set of French elisions is `l'`, `m'`, `t'`, `qu'`, `n'`, `s'`, `j'`, `d'`, `c'`, `jusqu'`, `quoiqu'`, `lorsqu'`, and `puisqu'`. You can update this by configuring the `french_elision` token filter. The following example request creates a new index named `french_texts` and configures an analyzer with `french_elision` filter:
```

> The default set of French elisions is `l'`, `m'`, `t'`, `qu'`, `n'`, `s'`, `j'`, `d'`, `c'`, `jusqu'`, `quoiqu'`, `lorsqu'`, and `puisqu'`. You can update this by configuring the `french_elision` token filter. The following example request creates a new index named `french_texts` and configures an analyzer with `french_elision` filter:

Reviewer 2: ```suggestion
The default set of French elisions is `l'`, `m'`, `t'`, `qu'`, `n'`, `s'`, `j'`, `d'`, `c'`, `jusqu'`, `quoiqu'`, `lorsqu'`, and `puisqu'`. You can update this by configuring the `french_elision` token filter. The following example request creates a new index named `french_texts` and configures an analyzer with the `french_elision` filter:
```

> The default set of French elisions is `l'`, `m'`, `t'`, `qu'`, `n'`, `s'`, `j'`, `d'`, `c'`, `jusqu'`, `quoiqu'`, `lorsqu'`, and `puisqu'`. You can update this by configuring the `french_elision` token filter. The following example request creates a new index named `french_texts` and configures an analyzer with the `french_elision` filter:

Reviewer 2: with "a" `french_elision` filter"?
